### PR TITLE
[Merged by Bors] - Document ZookeeperZnode

### LINKED ISSUE #301: User-facing docs for ZookeeperZnode
https://github.com/stackabletech/zookeeper-operator/pull/263#discussion_r768581996

### LINKED ISSUE #428: Document service discovery
As a user of services deployed by this operator I'd like to know how to discover its connection details.

It's done when

- documentation is available on all the objects (e.g. ConfigMaps) created by this operator and the circumstances under which they are created and
- documentation on the contents of the created objects is available.

NOTE: This ticket is part of an epic and autocreated for all our operators. It _might_ not apply to this operator in particular, in that case please comment and close
https://github.com/stackabletech/documentation/issues/86

### PR DESCRIPTION
**Pull request by teozkr**
## Description

Fixes #301, fixes #428

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by teozkr** [comment]
I've tried to cover both the motivation and basic usage, but I'd love it if someone could take a look at whether it makes sense.

I'm also not quite sure it makes sense to merge this without also covering discovery as a standalone topic. On the other hand, it might be better to get something out of the door than nothing...

File: docs/modules/ROOT/pages/znodes.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Apache ZooKeeper organizes all data into a hierarchical system of https://zookeeper.apache.org/doc/r3.7.0/zookeeperProgrammers.html#ch_zkDataModel[ZNodes],
+which act as both files (they can have data associated with them) and folders (they can contain other znodes) when compared to a traditional (POSIX-like) file system.
+
+In order to isolate different clients using the same ZooKeeper cluster, each client application should be assigned a unique root znode, which it can then organize
+as it sees fit.
+
+In the Stackable Operator for Apache ZooKeeper, we manage this using the `ZookeeperZnode` Kubernetes resource. The operator creates a new znode in the linked
+ZooKeeper cluster, as well as xref:discovery.adoc[] that are configured to restrict access to the configured ZNode. The operator _does not_ manage the contents
+of the ZNode.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/znodes.adoc:11]: Use either znode or ZNode.

File: docs/modules/ROOT/pages/znodes.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+CAUTION: The operator automatically deletes the znode from the ZooKeeper cluster if the Kubernetes `ZookeeperZnode` object is deleted. Recreating the
+`ZookeeperZnode` object will not restore access to the data.
+
+== Usage
+
+The `ZookeeperZnode` object should be created by user administering the system that will use the znode. For example,
```
Comment (maltesander) [inline docs/modules/ROOT/pages/znodes.adoc:18]:
```
The `ZookeeperZnode` object should be created by a user administering the system that will use the znode. For example,```

**Review by maltesander — COMMENTED** [review]
I like it.

File: docs/modules/ROOT/pages/znodes.adoc (hunk 1 above)
Comment (teozkr) [reply docs/modules/ROOT/pages/znodes.adoc:11]: Yeah.. the ZK documentation is wildly inconsistent here too. Maybe let's just standardize on ZNode?

File: docs/modules/ROOT/pages/znodes.adoc (hunk 2 above)
Comment (teozkr) [reply docs/modules/ROOT/pages/znodes.adoc]: Oops, fixed.

**Review by maltesander — APPROVED** [review]
LGTM.

**Pull request closed without merging**